test/e2e/e2e_ginkgo/metrics_e2eg_test.go

```diff
@@ -0,0 +1,125 @@
+		c          operatorclient.ClientInterface
+		crc        versioned.Interface
+		failingCSV v1alpha1.ClusterServiceVersion
+	)
+
+	g.It("Test Metrics Endpoint", func() {
+		c = newKubeClient()
+		crc = newCRClient()
+		failingCSV = v1alpha1.ClusterServiceVersion{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       v1alpha1.ClusterServiceVersionKind,
+				APIVersion: v1alpha1.ClusterServiceVersionAPIVersion,
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name: genName("failing-csv-test-"),
+			},
+			Spec: v1alpha1.ClusterServiceVersionSpec{
+				InstallStrategy: v1alpha1.NamedInstallStrategy{
+					StrategyName: v1alpha1.InstallStrategyNameDeployment,
+					StrategySpec: strategy,
+				},
+			},
+		}
+
+		g.By("creating a failing CSV")
+		cleanupCSV, err := createCSV(c, crc, failingCSV, testNamespace, false, false)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		defer cleanupCSV()
+
+		g.By("fetching the created CSV")
+		_, err = fetchCSV(crc, failingCSV.Name, testNamespace, csvFailedChecker)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By("Get metrics from pod and verify output")
+
+		rawOutput, err := getMetricsFromPod(c, getOLMPodName(c), operatorNamespace, "8081")
+		if err != nil {
+			e2e.Failf("Metrics test failed: %v\n", err)
+		}
+
+		// Verify metrics have been emitted for packageserver csv
+		o.Expect(rawOutput).To(o.ContainSubstring("csv_abnormal"))
+		o.Expect(rawOutput).To(o.ContainSubstring("name=\"" + failingCSV.Name + "\""))
+		o.Expect(rawOutput).To(o.ContainSubstring("phase=\"Failed\""))
+		o.Expect(rawOutput).To(o.ContainSubstring("reason=\"UnsupportedOperatorGroup\""))
+		o.Expect(rawOutput).To(o.ContainSubstring("version=\"0.0.0\""))
+		o.Expect(rawOutput).To(o.ContainSubstring("csv_succeeded"))
+		//log.Info(rawOutput)
+
+	})
+})
+
+func getOLMPodName(client operatorclient.ClientInterface) string {
+	listOptions := metav1.ListOptions{LabelSelector: "app=olm-operator"}
+	podList, err := client.KubernetesInterface().CoreV1().Pods(operatorNamespace).List(listOptions)
+	if err != nil {
+		e2e.Logf("Error %v\n", err)
+		e2e.Failf("Listing pods failed: %v\n", err)
+	}
+	if len(podList.Items) != 1 {
+		e2e.Failf("Expected 1 olm-operator pod, got %v", len(podList.Items))
+	}
+
+	podName := podList.Items[0].GetName()
+	e2e.Logf("Looking at pod %v in namespace %v", podName, operatorNamespace)
+	return podName
+
+}
+
+func getMetricsFromPod(client operatorclient.ClientInterface, podName string, namespace string, port string) (string, error) {
+	olmPod, err := client.KubernetesInterface().CoreV1().Pods(namespace).Get(podName, metav1.GetOptions{})
+	if err != nil {
+		return "", err
+	}
+	if len(olmPod.Spec.Containers) != 1 {
+		e2e.Failf("Expected only 1 container in olm-operator pod, got %v", len(olmPod.Spec.Containers))
+	}
+
+	var foundCert bool
+	var foundKey bool
+	// assuming -tls-cert and -tls-key aren't used anywhere else as a parameter value
+	for _, param := range olmPod.Spec.Containers[0].Args {
+		if param == "-tls-cert" {
```

Comment:
is the arg `--tls-cert` ?

Reply:
I believe not. `--tls-cert` fails the test